core/src/components/datetime/datetime.tsx

```diff
@@ -1262,6 +1236,58 @@ export class Datetime implements ComponentInterface {
        */
       this.activeParts = [];
     }
+
+    /**
+     * Only animate if:
+     * 1. We're using grid style (wheel style pickers should just jump to new value)
+     * 2. The month and/or year actually changed (otherwise there's nothing to animate to)
+     * 3. The datetime is visible (prevents animation when in collapsed datetime-button, for example)
+     */
+    const didChangeMonth = month !== workingParts.month || year !== workingParts.year;
+    const elIsVisible = el.offsetParent !== null;
+    if (animate && this.isGridStyle && didChangeMonth && elIsVisible) {
+      /**
+       * Tell other render functions that we need to force the
+       * target month to appear in place of the actual next/prev month.
+       * Because this is a State variable, a rerender will be triggered
+       * automatically, updating the rendered months.
+       */
+      this.forceRenderDate = { month, year, day };
+
+      /**
+       * Flag that we've started scrolling to the forced date.
+       * The resolve function will be called by the datetime's
+       * scroll listener when it's done updating everything.
+       * This is a replacement for making prev/nextMonth async,
+       * since the logic we're waiting on is in a listener.
+       */
+      const forceDateScrollingPromise: Promise<void> = new Promise((resolve) => {
+        this.resolveForceDateScrolling = resolve;
+      });
+
+      /**
+       * Animate smoothly to the forced month. This will also update
+       * workingParts and correct the surrounding months for us.
+       */
+      const targetMonthIsEarlier = isBefore(targetValue, workingParts);
+      targetMonthIsEarlier ? this.prevMonth() : this.nextMonth();
+      await forceDateScrollingPromise;
+      this.resolveForceDateScrolling = undefined;
+      this.forceRenderDate = null;
+    } else {
+      /**
+       * We only need to do this if we didn't just animate to a new month,
+       * since prevMonth/nextMonth call setWorkingParts for us.
+       */
+      this.setWorkingParts({
```

Comment:
Not really sure where to put this, so I'll put it here:

When the time popover is open and the value updates, pressing "Next" seems to jump over multiple months:

https://github.com/ionic-team/ionic-framework/assets/2721089/5172aa05-04d3-49e3-81b8-c7ff38751154

Reply:
Similar(?) problem, if you have the time picker open on that test and change the value dynamically to several months ahead, the next button is not actionable.

In the recording I show that with the time picker collapsed, the button works. After refreshing and opening the time picker, it does not. 

https://github.com/ionic-team/ionic-framework/assets/13732623/a309db00-93d7-4662-bcfd-cb009c462df5

Reply:
Both fixed in https://github.com/ionic-team/ionic-framework/pull/27806/commits/692a270e9a458efc4263aa0c2ddf0a65328583e1.